include/stdexec/execution.hpp

```diff
@@ -547,8 +650,24 @@ namespace stdexec {
 
   /////////////////////////////////////////////////////////////////////////////
   // [execution.receivers]
+
   template <class _Receiver>
     concept receiver =
+      // NOT TO SPEC:
+      // As we upgrade the receiver related entities from R5 to R7,
```

Comment:
Without additionally constraining the `receiver` concept to keep requiring receiver types provide a `get_env`, types like `no_env` became receivers. With CWG 2369, GCC-11 ended up instantiating more non-immediate context templates, and, for example, would try instantiating `when_all`'s `tag_invoke(connect_t, ...)`  with `_Receiver=no_env`. Clang and GCC-10 aren't bothered by this because (I'm guessing) CWG 2369 is not implemented there.

I don't have an explanation for why line 670 was needed for the code to compile with GCC-11 (it's also not needed for Clang) - it doesn't seem right, but I need to look further at that one.

Reply:
I think line 670 would go away once https://github.com/NVIDIA/stdexec/pull/752 is fixed